engines/scalac/src/main/scala/scala/macros/internal/engines/scalac/semantic/Abstracts.scala

```diff
@@ -163,8 +163,9 @@ trait Abstracts extends scala.macros.semantic.Mirrors { self: Universe =>
     }
 
     implicit class XtensionToGType(tpe: Type) {
-      def toGType: g.Type = tpe match {
+      def toGType(implicit m: Mirror): g.Type = tpe match {
         case gtpt: g.TypeTree => gtpt.tpe
+        case tname: Name => Symbol(tname.value)(m).tpe
         case _ => ???
       }
```

Comment:
@a-nigredo I think it's allright to keep `TypeTree(...)` in SyntaxAnalyzer. We just have to account for `TypeTree(...)` in our Type.XXX extractors. Wdyt?